lib/lowdown/connection.rb

```diff
@@ -83,8 +83,25 @@ class TimedOut < StandardError; end
     # @param  [Boolean] connect
     #         whether or not to immediately connect on initialization.
     #
-    def initialize(uri, ssl_context, connect = true)
+    # @param [lambda] socket_maker
+    #        a lambda takes uri and returns duck type of TCPSocket
+    #        e.g.:
+    #        ```
+    #        socket_maker = lambda do |uri|
+    #          Proxifier::Proxy('http://127.0.0.1:3128').open \
+    #          uri.host, uri.port, nil, nil, Celluloid::IO::TCPSocket
+    #        end
+    #
+    #        connection_pool = Lowdown::Connection.pool \
+    #          size: 2,
+    #          args: [uri, cert.ssl_context, true, socket_maker]
+    #
+    #        client = Lowdown::Client.client_with_connection connection_pool, certificate: cert
+    #        ```
```

Comment:
This should probably use a YARD `@example` block: http://www.rubydoc.info/gems/yard/file/docs/Tags.md#example